apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/PageQueryBuilder.java

```diff
@@ -0,0 +1,43 @@
+package gov.cms.bfd.server.war.stu3.providers;
+
+import javax.persistence.EntityManager;
+import javax.persistence.Query;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+
+public final class PageQueryBuilder<T> {
+
+  public final Root<T> root;
+  public final CriteriaBuilder builder;
+  public final CriteriaQuery criteria;
```

Comment:
Should these be `private` for encapsulation sake and -- if they need to be exposed to users of this class -- complemented by getter methods?

Reply:
They certainly could be. I wasn't sure what the current idiomatic Java would be.

Reply:
Most methods on these objects modify them which is what drove me to use `public` in this case. For example reading `qb.critera.where(<predicate>)` it doesn't surprise me that `qb.criteria` is now modified to include that `predicate`. Reading `qb.getCriteria().where(<predicate>)` it seems less obvious that the internal `qb.criteria` has been modified.

Do you think that holds up? Or is it just better to move to a getter pattern?

Reply:
Sounds like _builder-style_ accessor methods could be a good middle ground preserving encapsulation while avoiding awkwardness of getter-style naming? `qb.criteria().where(<predicate>)`